### PR TITLE
chore: Creates script to generate changelog entry

### PR DESCRIPTION
**Pull request by oarbusi**
## Description

Creates script to generate changelog entry. The script suggests the next PR number and uses changelog-entry command to simplify as much as possible the process 
Link to any related issue(s): CLOUDP-241729

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: .changelog/2127.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```
Comment (oarbusi) [inline .changelog/2127.txt]: generated using the new script

File: scripts/generate-changelog-entry.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+current_pr=$(curl -s "https://api.github.com/repos/mongodb/terraform-provider-mongodbatlas/issues?state=all&per_page=1" | jq -r ".[].number")
+next_pr=$((current_pr + 1))
+
+echo "==> What is the new changelog entry? Suggested is $next_pr (next PR number)"
+read -r next_pr
+
+changelog-entry -pr "$next_pr" -dir ".changelog"
+
+echo
+echo "Successfully created $next_pr. Don't forget to commit it and open the PR!"
```
Comment (lantoli) [inline scripts/generate-changelog-entry.sh]: new line at the end

File: .changelog/2127.txt (hunk 1 above)
Comment (lantoli) [inline .changelog/2127.txt]: great to have an example, but not sure if this particular PR should be in the changelog as it doesn't have a customer-face feature

**Review by lantoli — APPROVED** [review]

File: .changelog/2127.txt (hunk 1 above)
Comment (oarbusi) [reply .changelog/2127.txt]: Probably not. I can remove it before merging.

Comment (AgustinBettati) [reply .changelog/2127.txt]: agreed, lets make sure to remove before the merge

File: scripts/generate-changelog-entry.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Comment (AgustinBettati) [inline scripts/generate-changelog-entry.sh:1]: Could we include a make command to invoke the script?

File: scripts/generate-changelog-entry.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  exit 1
+fi
+
+current_pr=$(curl -s "https://api.github.com/repos/mongodb/terraform-provider-mongodbatlas/issues?state=all&per_page=1" | jq -r ".[].number")
+next_pr=$((current_pr + 1))
+
+echo "==> What is the new changelog entry? Suggested is $next_pr (next PR number)"
+read -r next_pr
+
+changelog-entry -pr "$next_pr" -dir ".changelog"
```
Comment (AgustinBettati) [inline scripts/generate-changelog-entry.sh]: Our guidelines [define a limited set of header types](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2123/files#diff-57e0bba85dc98868e7f9f864235d7316522b9512271bd6e4394fd3934b27bb0eR45), can we make use of the [allowed types flag](https://github.com/hashicorp/go-changelog/tree/main/cmd/changelog-entry#customizing-the-allowed-types) so we dont use any additional header types?

File: scripts/generate-changelog-entry.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+if ! command -v changelog-entry &> /dev/null
+then
+  echo "changelog-entry not be found"
+  exit 1
+fi
+
+current_pr=$(curl -s "https://api.github.com/repos/mongodb/terraform-provider-mongodbatlas/issues?state=all&per_page=1" | jq -r ".[].number")
+next_pr=$((current_pr + 1))
+
+echo "==> What is the new changelog entry? Suggested is $next_pr (next PR number)"
```
Comment (AgustinBettati) [inline scripts/generate-changelog-entry.sh]:
```suggestion
echo "==> What PR number should be used for this changelog entry? Leave emtpy to use $next_pr (next PR number)"
```
not sure if something of this nature is possible, but would be nice so you dont have to copy the suggested value

Comment (oarbusi) [reply scripts/generate-changelog-entry.sh]: It is possible, done!

File: scripts/generate-changelog-entry.sh (hunk 4 above)
Comment (oarbusi) [reply scripts/generate-changelog-entry.sh]: Done!

File: scripts/generate-changelog-entry.sh (hunk 3 above)
Comment (oarbusi) [reply scripts/generate-changelog-entry.sh:1]: Done!

**Review by EspenAlbert — APPROVED** [review]
Nice 👍

File: scripts/generate-changelog-entry.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+if ! command -v changelog-entry &> /dev/null
+then
+  echo "changelog-entry not be found"
+  exit 1
+fi
+
+current_pr=$(curl -s "https://api.github.com/repos/mongodb/terraform-provider-mongodbatlas/issues?state=all&per_page=1" | jq -r ".[].number")
+next_pr=$((current_pr + 1))
+
+echo "==> What PR number should be used for this changelog entry? Leave emtpy to use $next_pr (next PR number)"
```
Comment (AgustinBettati) [inline scripts/generate-changelog-entry.sh]:
```suggestion
echo "==> What PR number should be used for this changelog entry? Leave empty to use $next_pr (next PR number)"
```

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for the followups

**Pull request merged**